src/internal/task/queue.go

```diff
@@ -0,0 +1,88 @@
+package task
+
+// Queue is a FIFO container of tasks.
+// The zero value is an empty queue.
+type Queue struct {
+	head, tail *Task
+}
+
+// Push a task onto the queue.
+func (q *Queue) Push(t *Task) {
+	if q.head != nil {
```

Comment:
```
panic("runtime: pushing a task to a queue with a non-nil Next pointer")
```
I am not quite sure why this is treated as an error. `Push` is in charge of clearing the next field, and tasks are passed in with non-nil next pointers (mostly when transferring from another source such as a stack).

What other assertions did you have in mind?

Reply:
I made sure to always clear the next pointer when there was no next task. Even though that might cause a small performance penalty, it makes sure there are invariants that can be checked. Also, it might help the GC if the `Next` goroutine is not referenced anymore (for example, it's blocked).

I have found bugs in the past by checking this invariant. The last was a memory corruption while I was porting the tasks scheduler over to RISC-V (still not finished, I'll wait until after this PR is merged).

Reply:
The only other assert I can find is `panic("runtime: addSleepTask: expected next task to be nil")` which I think is really the same check.